[blog] Add blog post for high-level overview of all MUI products (#34325)
mui/material-ui: docs/pages/blog/mui-product-comparison.md

- Community: Over 2,500 open-source contributors have made this library what it is today.

#### Ideal use cases

- User interfaces that adhere closely to Material Design
- Internal/admin tools
- Dev teams that need to ship features in hours rather than weeks

### Joy UI

Joy UI is an open-source React component library that implements MUI's own in-house design principles.
It's intended to serve as an alternative to Material UI for designs that don't adhere to Material Design specifications.

Try Joy UI if you appreciate the comprehensiveness and reliability of Material UI, but don't need all of the additional baggage that comes along with Material Design.

Get started in the [Joy UI docs](/joy-ui/getting-started/overview/).

#### Key features

- Modern design: Joy UI's default styles align better with modern sensibilities than Material Design's more conservative take.
- Flexibility: Customize with ease, and leverage the power of CSS variables to ensure consistency when making pixel-perfect adjustments.

#### Ideal use cases

- Projects that don't require Material Design

### MUI Base

> MUI Base is a library of headless ("unstyled") React UI components.

Reviewer: ```suggestion
MUI Base is a library of headless ("unstyled") React UI components ands hooks.
```

These components were extracted from Material UI, and are now available as a standalone package. They feature the same robust engineering but without any default styling solution or theme.

MUI Base includes prebuilt components with production-ready functionality, along with low-level hooks for transferring that functionality to other components.

Get started in the [MUI Base docs](/base/getting-started/overview/).

#### Key features

#### Ideal use cases

- Implementing fully custom designs

### MUI System

MUI System is a set of CSS utilities to help you build custom designs more efficiently when working with MUI component libraries like Material UI, Joy UI, and MUI Base.

The System gives you a set of flexible, generic wrapper components like Box and Container that can be quickly customized using the sx prop.
This prop lets you define styles directly within the components themselves, rather than creating bulky and redundant const definitions with styled-components.

> It also gives you direct access to your theme's custom design tokens to ensure consistency in one-off styles. Learn more on the sx prop page.

Reviewer: ```suggestion
It also gives you direct access to your theme's custom design tokens to ensure consistency in one-off styles trough the sx prop. Learn more about it [here](/system/getting-started/the-sx-prop/).
```

What began as an open-source React UI library has evolved over time into a full-fledged startup with a whole suite of related products.

Material UI is the flagship; but it's also the gateway to MUI's ever-expanding ecosystem of UI tools.

Though our roots are in [Material Design](https://material.io/), we're branching out well beyond those constraints these days to deliver a wider range of tools for developers to ship new features faster.

Our primary offerings fall into two product lines: Core and X.
MUI Core contains our foundational component libraries (like Material UI), while MUI X offers components that are significantly more complex (like the Data Grid).

We're also in the early stages of developing a low-code internal tool builder called [MUI Toolpad](https://mui.com/toolpad/), in which you can build with every Core and X component in a drag-and-drop interface.

Read on for more details on each of our products.

## MUI Core

The Core is MUI's foundational product line.
It grew out of Material UI, and that library's legacy lives on in the name of the repo on GitHub: [`mui/material-ui`](https://github.com/mui/material-ui).

> But this repo contains much more than just Material UI these days.

Reviewer 2: I'm not sold on the use of these adverbs. When I see "much, "lot", "just", or equivalent then it quickly becomes obvious where the person is trying to orient the discussion. It often feels like putting his own judgment above how facts could resonate with the people the message is written for. This get me to this alternative:

```suggestion
But this repository now contains more than Material UI.
```

We could also maybe expand on the product direction taken, I think that it's as simple as:

```suggestion
But this repository now contains more than Material UI. We have disassembled Material UI to expose its best parts as isolated open-source projects.
```

- **Community:** Over 2,500 open-source contributors have made this library what it is today.

#### Ideal use cases

- User interfaces that adhere closely to Material Design
- Internal/admin tools
- Dev teams that need to ship features in hours rather than weeks

### Joy UI

Joy UI is an open-source React component library that implements MUI's own in-house design principles.
It's intended to serve as an alternative to Material UI for designs that don't adhere to Material Design specifications.

Try Joy UI if you appreciate the comprehensiveness and reliability of Material UI, but don't need all of the additional baggage that comes along with Material Design.

Get started in the [Joy UI docs](/joy-ui/getting-started/overview/).

#### Key features

- **Innovative design:** Free from the constraints of Material Design, Joy UI is where we can innovate and experiment with fresh new ideas in design, UX, and DX.
- **Flexibility:** Customize with ease, and leverage the power of CSS variables to ensure consistency when making pixel-perfect adjustments.

#### Ideal use cases

- Projects that don't involve Material Design

> - Designs that would benefit from less opinionated defaults when customizing

Reviewer 3: ```suggestion
- Design systems that would benefit from less opinionated defaults when customizing
```

I would like to add a point about the design system here but not sure if it should be a bullet. My intention is around "If you are building a fresh design system, you should try Joy UI."

It also gives you direct access to your theme's custom design tokens to ensure consistency in one-off styles.

Get started in the [MUI System docs](/system/getting-started/overview/).

#### Key features

- **Faster prototyping:** The `sx` prop lets you apply styles directly to a component by writing the CSS you already know.
  This makes it a great tool for quickly assembling a prototype or styling a one-off component that doesn't need to be reused.
- **Write less code:** `sx` lets you avoid writing unnecessary styled-components code, and as such it can replace dozens of lines of code.
  Check out [Why use MUI System?](/system/getting-started/usage/#why-use-mui-system) for an example.

#### Ideal use cases

- Quickly prototyping custom styles in Material UI, Joy UI, or MUI Base
- Creating one-off custom components

## MUI X

MUI X is a collection of advanced UI components, including the Data Grid and the Date and Time Pickers.

These components are significantly more complex than those found in the MUI Core libraries.
They feature advanced functionality for data-rich applications and a wide range of other use cases.

> MUI X is open core—base components are MIT-licensed, while more advanced features require a Pro or Premium commercial license.

Reviewer 4: ```suggestion
MUI X is open-core. Base components are MIT-licensed, while more advanced features require a Pro or Premium commercial license.
```

See [Licensing](/x/introduction/licensing/) for details.

### Data Grid

> The MUI X Data Grid is a fast and extensible React data table that's built to handle massive sets of data for management and analysis.

Reviewer 4: ```suggestion
The MUI X Data Grid is a fast and extensible React data table with advanced features for power users and complex use cases in data management and analysis.
```

> It features an intuitive API for real-time updates and accessibility, as well as theming and custom templates.

Reviewer 4: ```suggestion
It features intuitive UI and API to display different views of a dataset, with support for real-time updates, accessibility, theming, and custom templates.
```

#### Key features

#### Ideal use cases

- data-rich applications
- data analytics tools
- admin tools

### Date and Time Pickers

> The MUI X Date and Time Pickers are a collection of React UI components that let users select dates and times from dialogs and text field dropdowns.

Reviewer 4: ```suggestion
The MUI X Date and Time Pickers are a collection of React UI components that let users select dates and times from dialogs and text fields.
```

---
title: An introduction to the MUI ecosystem
description: MUI is more than just Material UI. Consider Joy UI, MUI Base, MUI X, and Toolpad for your next project.
date: 2022-10-24T00:00:00.000Z
authors: ['samuelsycamore']
card: true
tags: ['Product']
---

<img src="/static/blog/mui-product-comparison/ecosystem.png" style="width: 692px; margin-bottom: 24px; aspect-ratio: 173/80" alt="Logos for all the high-level MUI products." />

MUI is so much more than Material UI!

You'll be forgiven if you thought MUI was merely shorthand for our most popular product.

> What began as an open-source React UI library has evolved over time into a full-fledged startup with a whole suite of related products.

Reviewer 2: Do describe a stronger expansion of scope?

```suggestion
What began as an open-source React UI library that implements Material Design has evolved over time into a full-fledged startup with a whole suite of related products to build UI faster.
```

Author: I think these details are made clear enough in the next couple lines.

Reviewer 2: What got me confused is that "what started at X" but we are still mostly X.

Author: I think I see the source of the confusion, which is that Material UI didn't _become_ MUI—instead, MUI _grew out of_ Material UI. So it's not really correct to say that the "library" has evolved into a "startup."

How about this instead? I think it conveys the same idea more clearly, without repeating what's already stated elsewhere in the intro.

```suggestion
In fact, MUI as an organization was founded to meet the growing needs of the Material UI community, and is now responsible for a whole suite of related products.
```

The Core is MUI's foundational product line.
It grew out of Material UI, and that library's legacy lives on in the name of the repo on GitHub: [`mui/material-ui`](https://github.com/mui/material-ui).

But this repo contains much more than just Material UI these days.
We've carefully deconstructed this library to expose its best parts as isolated open-source projects.
More recent additions include Joy UI and MUI Base, as well as our in-house styling solution, MUI System.

MUI Core is open-source, and we invite you to contribute wherever you see fit!

### Material UI

Material UI is an open-source React component library that implements Google's Material Design.
It includes a comprehensive collection of prebuilt components that are ready for use in production right out of the box.

Material UI is beautiful by design and features a suite of customization options that make it easy to implement your own custom design system on top of our components.

Get started in the [Material UI docs](/material-ui/getting-started/overview/).

#### Key features

> - **Material Design:** Your app will look and feel excellent by default, thanks to our meticulous implementation of Material Design.

Reviewer 2: Maybe to give more context.

```suggestion
- **Material Design:** Your app will look and feel excellent by default, thanks to our meticulous implementation of Material Design. It currently implements Material Design 2.
```

Get started in the [MUI System docs](/system/getting-started/overview/).

#### Key features

- **Faster prototyping:** The `sx` prop lets you apply styles directly to a component by writing the CSS you already know.
  This makes it a great tool for quickly assembling a prototype or styling a one-off component that doesn't need to be reused.
- **Write less code:** `sx` lets you avoid writing unnecessary styled-components code, and as such it can replace dozens of lines of code.
  Check out [Why use MUI System?](/system/getting-started/usage/#why-use-mui-system) for an example.

#### Ideal use cases

- Quickly prototyping custom styles in Material UI, Joy UI, or MUI Base
- Creating one-off custom components

## MUI X

<img src="/static/blog/mui-product-comparison/mui-x.png" style="width: 692px; margin-bottom: 24px; aspect-ratio: 173/75" loading="lazy" alt="Diagram with all the MUI X components." />

MUI X is a collection of advanced UI components, including the Data Grid and the Date and Time Pickers.

These components are significantly more complex than those found in the MUI Core libraries.

> They feature advanced functionality for data-rich applications and a wide range of other use cases.

Reviewer 2: Maybe to explain why MUI X exists in the first place:

```suggestion
They feature advanced functionality for data-rich applications and a wide range of other use cases.
While these components provide native integration with MUI Core, they are separated so they can focus on deep integration with engineers' custom design systems.
```

MUI X is open-core. Base components are MIT-licensed, while more advanced features require a Pro or Premium commercial license.
See [Licensing](/x/introduction/licensing/) for details.

### Data Grid

The MUI X Data Grid is a fast and extensible React data table with advanced features for power users and complex use cases in data management and analysis.
It features an intuitive UI and corresponding API to display different views of a data set, with support for real-time updates, accessibility, theming, and custom templates.

#### Key features

- **Sophisticated data analysis:** Edit, filter, sort, aggregate, and export your data with ease. MUI X's Data Grid is one of the most feature-rich grids available on the market today.
- **Intuitive customization:** Customizing the Data Grid's styles and behaviors works just like you'd expect if you're familar with Material UI—no new concepts to master along the way.
- **Accessibility:** It is notoriously difficult to optimize a data grid's accessibility, which is precisely why we make it a top priority—we obsess over these details so you don't have to.

#### Ideal use cases

- Applications that feature massive data sets

> - Data analytics tools

Reviewer 2: One of the key value is not to have to open Excel to do a pivot on the data or plot a chart to analyse the results.

```suggestion
- Data analytics tools inside your app, reducing the need to go to Excel
```

Author: That's a good detail!

- Apps built with MUI Core libraries that need advanced functionality

### Date and Time Pickers

The MUI X Date and Time Pickers are a collection of React UI components that let users select dates and times from dialogs and text fields.
Components include the Date Picker, Time Picker, Date Range Picker, and Date Time Picker, with more on the way.

#### Key features

**Third-party date library support:** The Date and Time Pickers include support for many of the most popular date libraries, so you can pick whichever you prefer.
**Localization:** The Pickers are optimized for use anywhere in the world, with support for all variations on date and time formatting.

#### Ideal use cases

- Forms requiring date and time input from the user
- Apps built with MUI Core libraries that need advanced functionality

## MUI Toolpad

<img src="/static/blog/mui-product-comparison/mui-toolpad.png" style="width: 692px; margin-bottom: 24px;" loading="lazy" alt="Small screenshot of MUI Toolpad's interface." />

> Toolpad is a self-hosted low-code internal tools builder designed to extend MUI's suite of React components.

Reviewer 2: The focus is more around admin, which can be dashboard, internal tools.

```suggestion
Toolpad is a self-hosted low-code admin builder designed to extend MUI's suite of React components.
```

## MUI X

<img src="/static/blog/mui-product-comparison/mui-x.png" style="width: 692px; margin-bottom: 24px; aspect-ratio: 173/75;" loading="lazy" alt="Diagram with all the MUI X components." />

MUI X is a collection of advanced UI components, including the Data Grid and the Date and Time Pickers.

These components are significantly more complex than those found in the MUI Core libraries.
They feature advanced functionality for data-rich applications and a wide range of other use cases.
While X components natively integrate with MUI Core, they are kept separate so you can more easily implement custom design systems with them.

MUI X is open-core. Base components are MIT-licensed, while more advanced features require a Pro or Premium commercial license.
See [Licensing](/x/introduction/licensing/) for details.

### Data Grid

The MUI X Data Grid is a fast and extensible React data table with advanced features for power users and complex use cases in data management and analysis.
It features an intuitive UI and corresponding API to display different views of a data set, with support for real-time updates, accessibility, theming, and custom templates.

#### Key features

> - **Sophisticated data analysis:** Edit, filter, sort, aggregate, and export your data with ease. MUI X's Data Grid is one of the most feature-rich grids available on the market today.

Reviewer 4: I think we need to present we also support basic use cases with the Data Grid as well, hence the suggestion for a new line. Feel free to iterate over it.

```suggestion
- **Data management:** Create, retrieve, update and delete your data with ease. 
- **Sophisticated data analysis:** Pin rows and columns, group and aggregate values, export your view of the data to other formats. MUI X's Data Grid is one of the most feature-rich grids available on the market today.
```

- **Intuitive customization:** Customizing the Data Grid's styles and behaviors works just like you'd expect if you're familar with Material UI—no new concepts to master along the way.
- **Accessibility:** It's notoriously difficult to optimize a data grid's accessibility, which is precisely why we make it a top priority—we obsess over these details so you don't have to.

#### Ideal use cases

- Applications that feature massive data sets
- Data analytics tools inside your app, reducing the need to jump to Excel
- Apps built with MUI Core libraries that need advanced functionality

### Date and Time Pickers

The MUI X Date and Time Pickers are a collection of React UI components that let users select dates and times from dialogs and text fields.
Components include the Date Picker, Time Picker, Date Range Picker, and Date Time Picker, with more on the way.

#### Key features

> - **Third-party date library support:** The Date and Time Pickers include support for many of the most popular date libraries, so you can pick whichever you prefer.

Reviewer 4: To include the fields.

```suggestion
- **Third-party date library support:** All components include support for many of the most popular date libraries, so you can pick whichever you prefer.
```

> - **Localization:** The Pickers are optimized for use anywhere in the world, with support for all variations on date and time formatting.

Reviewer 4: We've been working a lot on usability, and although most of the improvements are going on v6, considering that the content of this blogpost might be referenced for a while, I think it's fair to add a point about it.

```suggestion
- **Localization:** The Pickers are optimized for use anywhere in the world, with support for all variations on date and time formatting.
- **Refined UX**: Adopt the best practices in terms of usability for your date and time inputs.
```

Reviewer 5: Should we link to some references when talking about best practices? What are our benchmarks? Are we testing ourselves or picking up from somebody else? Suggesting these answers might be good!

#### Ideal use cases

> - Forms requiring date and time input from the user

Reviewer 4: ```suggestion
- Forms requiring date and time input from the user.
```

> - Apps built with MUI Core libraries that need advanced functionality

Reviewer 4: Trying to break it down into different examples.

```suggestion
- Booking apps using date and time ranges.
- Enterprise apps using advanced date and time validation and constraints.
- Apps built with MUI Core libraries using date and time functionality.
```
